spec/std/http/headers_spec.cr

```diff
@@ -206,4 +206,18 @@ describe HTTP::Headers do
     headers.add?("foobar", invalid_value).should be_false
     headers.add?("foobar", [invalid_value]).should be_false
   end
+
+  it "raises when already sent (#8712)" do
+    headers = HTTP::Headers.new
+    headers.mark_as_sent!
+
+    expect_raises(ReadOnlyError) { headers["Foo"] = "Bar" }
```

Comment:
Could you add a spec that accessing a header still works after `mark_as_sent!`?

Reply:
Done!